### PR TITLE
Update to cats-effect 3

### LINKED ISSUE #51: use TestContext in SttpOauth2ClientCredentialsCatsBackendSpec instead of sleeps
Note that you can use `TestContext()` to do a sleep without actually waiting in the tests. I would recommend to do that, but it's not a blocker for now (with just a couple tests). See https://typelevel.org/cats-effect/testing/#testing-concurrency for some info

_Originally posted by @kubukoz in https://github.com/ocadotechnology/sttp-oauth2/pull/46#discussion_r590823934_

### PR DESCRIPTION
**Pull request by kubukoz**
Probably want to keep this open for a while, until you're ready to move to CE3. This also resolves #51.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -57,12 +57,12 @@ ThisBuild / githubWorkflowEnv ++= List("PGP_PASSPHRASE", "PGP_SECRET", "SONATYPE
...
 val Versions = new {
   val catsCore = "2.5.0"
-  val catsEffect = "2.3.1"
+  val catsEffect = "3.0.2"
   val circe = "0.13.0"
   val kindProjector = "0.11.3"
   val monix = "3.3.0"
   val scalaTest = "3.2.7"
-  val sttp = "3.2.3"
+  val sttp = "3.3.0-RC1"
```
Comment (kubukoz) [inline build.sbt:65]: should be updated to a stable version before merge

Comment (majk-p) [reply build.sbt]: done

**Comment by kubukoz** [comment]
I made a milestone of this: `0.9.0-ce3-M1`, if someone wants to try it out

**Comment by bwiercinski** [comment]
I think this can be closed since the changes are not related to main branch anymore. I will create a new PR.

**Pull request closed without merging**